### ISSUE TITLE
Feature or Option to Display One Definition at a Time

### ISSUE THREAD

**Issue by Mo-Zone** [hilverd/glossary-page-template#6]
Remarkable template Hilverd! We are currently exploring ways to rebuild our [Mathematics Glossary](https://www.learnalberta.ca/content/memg/2_A/index.html) on the Alberta Education's LearnAlberta.ca. Each term definition could have a number of media types, like images, video and HTML5 interactive pieces displayed in the definition display frame. If we incorporated a version of your template it would need to load just the selected definition into the frame where you currently show all the definition cards. I would love to hear your thoughts? Here is a bit of a mockup. Thanks so much!

<img width="1383" alt="one_click_glossary" src="https://user-images.githubusercontent.com/123698218/215135716-97e96aa2-4eff-43cc-bad7-079944a11103.png">

**Comment by hilverd** [comment]
Thanks for considering this template for your project!

Unfortunately what you are describing probably won't be possible any time soon. At the moment the template only supports plain text definitions consisting of a single (possibly long) paragraph for each definition. Most glossaries I have seen tend to have only brief definitions and I wanted to discourage overly long entries.

On the other hand, I am currently attempting to add Markdown support to try and provide some enhancements (e.g. links, images, ordered/unordered lists) that have been asked for. That would then also allow for longer entries with multiple paragraphs, and the existing layout (showing cards for all the items) probably wouldn't work so well -- so it might have to look more like the mockup you provided. Maybe I could still keep the cards but truncate them to a certain maximum length, and have a "View full item" button for longer ones (I'm just speculating at this point).

Adding Markdown support is going rather slowly so it's really difficult to predict if/when any of this might be possible.

**Comment by Mo-Zone** [comment]
Thank you for the quick reply Hilverd! Yes it is quite a departure from the current template, so I completely understand. Adding links and graphics to the current build would be awesome and would also fill our needs. 

I had floated this idea to the team before,  because there is something advantageous about having the all the cards available with just the definition and then a full view button, just as you describe. It provides a quick look at the basic definition. A lot of times that's all the user is looking for.

I was able to load up the empty template on my Mac and installed Node.JS so I could use the add terms function. Not bad for an old designer! Such a great feature which would allow our teachers to add content!

Thanks again and I look forward to future enhancements. Cheers, Maurice

**Comment by hilverd** [comment]
[Version 2.0.0 of the template](https://github.com/hilverd/glossary-page-template/releases/tag/v2.0.0) adds a few improvements. Maybe this is a step in the right direction for the needs you described? See e.g. the [Python 3.11.2 Glossary](https://glossary.page/template/examples/python-3.11.2-glossary) for an example.

I ended up making longer items scrollable and adding a settings option to make the cards wider. I guess I could still add a "View full item" button if that seems like a good idea.

On the other hand, I've had another quick look at the existing [Mathematics Glossary](https://www.learnalberta.ca/content/memg/2_A/index.html) and there are so many items with images that I suppose it might not be a good idea to have them all on a single HTML page...

**Comment by Mo-Zone** [comment]
Thank you Hilverd, this is fantastic. We are going to try out some edits. I have the High School Math Glossary started in version 1. So I just need to replace the .js and .css URLs in the html file, yeah? Cheers!

**Comment by hilverd** [comment]
Yes, if you update the version in the CSS and JS URLs then (when editing) you should see a settings menu where you can turn on Markdown support, among other things. I've tried to describe the steps in the [release notes](https://github.com/hilverd/glossary-page-template/releases/tag/v2.0.0).

If you run into any problems please let me know.
